### PR TITLE
PR -> Brazilian bank codes #367

### PR DESCRIPTION
**Pull request by Vido**
**Please replace these instructions with a description of your change. The
'New Fields Only' section should be removed if your pull request
doesn't add any new fields.**

Thanks for your contribution!

A checklist is included below which helps us keep the code contributions
consistent and helps speed up the review process. You can add additional
commits to your pull request if you haven't met all of these points on your
first version.

**All Changes**

- [x] Add an entry to the docs/changelog.rst describing the change.

- [x] Add an entry for your name in the docs/authors.rst file if it's not
      already there.

**New Fields Only**

- [x] Prefix the country code to all fields.

- [x] Field names should be easily understood by developers from the target
      localflavor country. This means that English translations are usually
      not the best name unless it's for something standard like postal code,
      tax / VAT ID etc.

- [x] Prefer '<country code>PostalCodeField' for postal codes as it's
      international English; ZipCode is a term specific to the United
      States postal system.

- [x] Add meaningful tests. 100% test coverage is not required but all
      validation edge cases should be covered.

- [ ] Add `.. versionadded:: <next-version>` comment markers to new
      localflavors.

- [ ] Add documentation for all fields.

### CODE REVIEW

File: tests/test_br/test_br.py [hunk 1]
```diff
@@ -201,6 +201,21 @@ def test_BRStateChoiceField(self):
...
+            '001': '001',
+        }
+        invalid = {
+            '9999': error_maxlength,
+            '1': error_maxlength,
+            'BCO': error_invalid,
+        }
+        self.assertFieldOutput(BRBankChoiceField, valid, invalid)
+
+
```
Comment (benkonrath) [inline tests/test_br/test_br.py]: This extra empty line should be removed.

File: localflavor/br/models.py [hunk 2]
```diff
@@ -69,3 +70,19 @@ def __init__(self, *args, **kwargs):
         kwargs['max_length'] = 9
         super().__init__(*args, **kwargs)
         self.validators.append(validators.BRPostalCodeValidator())
+
+
+class BRBankField(CharField):
+    """A model field for Brazilian Banks (COMPE code)."""
```
Comment (benkonrath) [inline localflavor/br/models.py]: Please add a `versionadded` tag in the comment.

File: localflavor/br/forms.py [hunk 3]
```diff
@@ -178,3 +179,34 @@ def clean(self, value):
             raise ValidationError(self.error_messages['invalid'])
 
         return orig_value
+
+
+class BRBankChoiceField(Field):
+    """
+    A choice field that uses a list of Brazilian Banks as its choices.
+    - Uses COMPE code as id
+    """
```
Comment (benkonrath) [inline localflavor/br/forms.py]: Please add a `versionadded` tag in the comment.

File: tests/test_br/test_br.py [hunk 4]
```diff
@@ -250,3 +265,11 @@ def test_BRPostalCodeField(self):
...
         self.assertEqual(instance.description, new_instance.description)
         self.assertEqual(instance.validators, new_instance.validators)
+
+    def test_BRBankField(self):
+        instance = models.BRBankField()
+        name, path, args, kwargs = instance.deconstruct()
+        new_instance = models.BRBankField(*args, **kwargs)
+        self.assertEqual(instance.max_length, new_instance.max_length)
+        self.assertEqual(instance.description, new_instance.description)
+        self.assertEqual(instance.validators, new_instance.validators)
```
Comment (benkonrath) [inline tests/test_br/test_br.py]: You should actually test the functionality on the model field; for instance, checking that the choices field is set correctly. Checking the validators like the other fields isn't required since you're not setting a validator.

File: localflavor/br/forms.py [hunk 5]
```diff
@@ -178,3 +179,34 @@ def clean(self, value):
             raise ValidationError(self.error_messages['invalid'])
 
         return orig_value
+
+
+class BRBankChoiceField(Field):
```
Comment (benkonrath) [inline localflavor/br/forms.py]: I think it's better use `ChoiceField` which would really simplify the code. I think doing something like this should work:

```
class BRBankChoiceField(ChoiceField):
    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        self.choices = BR_BANK_CHOICES
```

You can override the `invalid_choice` key in `default_error_messages` if you want a different error message.

**Comment by benkonrath** [comment]
I'm closing this PR due to inactivity. Feel free to re-open it and implement / reply to the requested changes if you want to continue with this contribution.
